### PR TITLE
Header footer templates

### PR DESCRIPTION
**Pull request by anjmao**
closes https://github.com/ng-select/ng-select/issues/184

### CODE REVIEW

File: demo/app/examples/custom-templates.component.ts [hunk 1]
```diff
@@ -60,12 +84,29 @@ export class SelectWithTemplatesComponent {
 
     cities2 = JSON.parse(JSON.stringify(this.cities));
     cities3 = JSON.parse(JSON.stringify(this.cities));
+    cities4 = JSON.parse(JSON.stringify(this.cities));
```
Comment (varnastadeus) [inline demo/app/examples/custom-templates.component.ts]: why not just slice?
```javascript
cities4 = this.cities.slice();
```

File: src/ng-select/ng-select.component.scss [hunk 2]
```diff
@@ -262,6 +262,16 @@ $color-selected: #f5faff;
             }
         }
     }
+
+    .ng-dropdown-header {
+        border-bottom: 1px solid #ccc;
```
Comment (varnastadeus) [inline src/ng-select/ng-select.component.scss]: not sure if predefined border is good idea

**Comment by varnastadeus** [comment]
good job! 👍

**Comment by arun-maddheshia** [comment]
@varnastadeus please merge it if its good, I am waiting for this update :+1:

**Comment by anjmao** [comment]
@arun-maddheshia Check it out and let me know if it solves you problem.

**Pull request merged**